core/common/src/MapImagerySettings.ts

```diff
@@ -6,47 +6,221 @@
+      case "BingProvider":
+      default:
+        label = `Bing Maps`;
+        break;
+
+      case "MapBoxProvider":
+        label = `MapBox`;
+        break;
+    }
+    return label;
+  }
+
+  public label() {
+    return  `${this.providerLabel()}: ${this.mapTypeName()}`;
+  }
+
+  public toMapSettings(): MapLayerSettings {
+    let formatId: string, url: string;
+    switch (this.providerName) {
+      case "BingProvider":
+      default:
+        formatId = "BingMaps";
+
+        let imagerySet;
+        switch (this.mapType) {
+          case BackgroundMapType.Street:
+            imagerySet = "Road";
+            break;
+          case BackgroundMapType.Aerial:
+            imagerySet = "Aerial";
+            break;
+          case BackgroundMapType.Hybrid:
+          default:
+            imagerySet = "AerialWithLabels";
+            break;
+        }
+        url = `https://dev.virtualearth.net/REST/v1/Imagery/Metadata/${imagerySet}?o=json&incl=ImageryProviders&key={bingKey}`;
+        break;
+
+      case "MapBoxProvider":
+        formatId = "MapboxImagery";
+        switch (this.mapType) {
+          case BackgroundMapType.Street:
+            url = "https://api.mapbox.com/v4/mapbox.streets/";
+            break;
+          case BackgroundMapType.Aerial:
+            url = "https://api.mapbox.com/v4/mapbox.satellite/";
+            break;
+          case BackgroundMapType.Hybrid:
+            url = "https://api.mapbox.com/v4/mapbox.streets-satellite/";
+            break;
+        }
+        break;
+    }
+    const name = this.label();
+    return MapLayerSettings.fromJSON({ name, formatId, url,
+      transparentBackground: false,
+      isBase: true,
+      visible: this.visible,
+      transparency: this.transparency })!;
+  }
+
+  public clone(changedProps: BackgroundMapProviderProps): BackgroundMapProvider {
+    if (undefined === changedProps)
+      return this;
+
+    const props = {
+      name: undefined !== changedProps.providerName ? changedProps.providerName : this.providerName,
+      providerData: {
+        mapType: undefined !== changedProps.providerData?.mapType ? changedProps.providerData.mapType : this.mapType,
+        visible: undefined !== changedProps.providerData?.visible ? changedProps.providerData.visible : this.visible,
+        transparency: undefined !== changedProps.providerData?.transparency ? changedProps.providerData.transparency : this.transparency,
+      },
+
+    };
+    const clone = BackgroundMapProvider.fromJSON(props)!;
+
+    return clone;
+  }
+
+  /** Construct from JSON, performing validation and applying default values for undefined fields. */
+  public static fromJSON(props: BackgroundMapProviderProps) {
+    return new BackgroundMapProvider(props);
+  }
+
+  /** Used to determine is a JSON struct matches BackgroundMapProviderProps  */
+  public static isMatchingProps(props: any): boolean {
+    return props?.hasOwnProperty("providerName") || props?.hasOwnProperty("providerData");
```

Comment:
Why insist an object that inherits a providerName or providerData property is not a BackgroundMapProviderProps?

Reply:
good catch, I will use the 'in' operator.